src/Controller/ForgotPasswordController.php

```diff
@@ -26,45 +26,48 @@
  */
 class ForgotPasswordController extends AbstractController
 {
-    private const SESSION_TOKEN_KEY = 'forgot_password_token';
-    private const SESSION_CAN_CHECK_EMAIL = 'forgot_password_check_email';
+    use ResetPasswordControllerTrait;
 
-    /** @TODO this value should be generated/retrieved from the config... */
-    private const LIFETIME_HOURS = 1;
+    private $helper;
+
+    public function __construct(ResetPasswordHelperInterface $helper)
+    {
+        $this->helper = $helper;
+    }
 
     /**
      * @Route("/request", name="app_forgot_password_request")
      */
-    public function request(Request $request, MailerInterface $mailer, ResetPasswordHelperInterface $passwordResetHelper): Response
+    public function request(Request $request, MailerInterface $mailer): Response
     {
         $form = $this->createForm(PasswordRequestFormType::class);
         $form->handleRequest($request);
 
         if ($form->isSubmitted() && $form->isValid()) {
-            return $this->processRequestForm($form, $request, $mailer, $passwordResetHelper);
+            return $this->processRequestForm($form, $request, $mailer);
         }
 
         return $this->render('forgot_password/request.html.twig', [
             'requestForm' => $form->createView(),
         ]);
     }
 
-    protected function processRequestForm(FormInterface $form, Request $request, MailerInterface $mailer, ResetPasswordHelperInterface $passwordResetHelper): RedirectResponse
+    protected function processRequestForm(FormInterface $form, Request $request, MailerInterface $mailer): RedirectResponse
     {
         $user = $this->getDoctrine()->getRepository(User::class)->findOneBy([
             'email' => $form->get('email')->getData(),
         ]);
 
         // Needed to be able to access next page, app_check_email
-        $request->getSession()->set(self::SESSION_CAN_CHECK_EMAIL, true);
+        $this->setCanCheckEmailInSession();
 
         // Do not reveal whether a user account was found or not.
         if (!$user) {
             return $this->redirectToRoute('app_check_email');
         }
 
         try {
-            $resetToken = $passwordResetHelper->generateResetToken($user);
+            $resetToken = $this->helper->generateResetToken($user);
         } catch (ResetPasswordExceptionInterface $e) {
             // TODO - make this better..
             $this->addFlash('error', sprintf(
```

Comment:
`error` should maybe be `reset_password_error` - in case the user is already using `error` in their app.

Reply:
Below, I'm dubious about having `getEmailTemplate()` as its own method - seems like we're splitting too far (not sure if this was done by the original generated code)

Reply:
- [ ] refactored error
- [ ] move getEmailTemplate into processRequestForm

Reply:
done